src/CMakeLists.txt

```diff
@@ -169,164 +157,45 @@ SET (HDRS
-
-	get_directory_property(BINDING_DEPENDENCIES DIRECTORY "Bindings" DEFINITION BINDING_DEPENDENCIES)
-
-	#clear file
-	file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/Bindings/BindingDependencies.txt)
-	foreach(dependency ${BINDING_DEPENDENCIES})
-		#write each dependency on a seperate line
-		file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/Bindings/BindingDependencies.txt "${dependency}\n")
-	endforeach()
-
-	set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES "Bindings.cpp Bindings.h")
-
-	list(APPEND SOURCE "${SRCS}")
-	list(APPEND SOURCE "${HDRS}")
-
-	# If building a windows version, but not using MSVC, add the resources directly to the makefile:
-	if (WIN32)
-		list(APPEND SOURCE "Resources/Cuberite.rc")
-	endif()
-else ()
-	# MSVC-specific handling: Put all files into one project, separate by the folders:
-
-	source_group(Bindings FILES "Bindings/Bindings.cpp" "Bindings/Bindings.h")
-
-	# Add all subfolders as solution-folders:
-	function(includefolder PATH)
-		FILE(GLOB FOLDER_FILES
-			"${PATH}/*.cpp"
-			"${PATH}/*.h"
-			"${PATH}/*.rc"
-			"${PATH}/*.pkg"
-		)
-		string(REPLACE "/" "\\" PROJECT_PATH ${PATH})
-		source_group("${PROJECT_PATH}" FILES ${FOLDER_FILES})
-	endfunction(includefolder)
-
-	foreach(folder ${FOLDERS})
-		add_subdirectory(${folder})
-		includefolder(${folder})
-
-		# Get all source files in this folder:
-		get_directory_property(FOLDER_SRCS DIRECTORY ${folder} DEFINITION SRCS)
-		foreach (src ${FOLDER_SRCS})
-			list(APPEND SOURCE "${folder}/${src}")
-		endforeach(src)
-
-		# Get all headers in this folder:
-		get_directory_property(FOLDER_HDRS DIRECTORY ${folder} DEFINITION HDRS)
-		foreach (hdr ${FOLDER_HDRS})
-			list(APPEND SOURCE "${folder}/${hdr}")
-		endforeach(hdr)
-
-		# Include this folder's CMakeLists.txt in the project:
-		list(APPEND SOURCE "${folder}/CMakeLists.txt")
-		source_group("${folder}" FILES "${folder}/CMakeLists.txt")
-	endforeach(folder)
-
-	list(APPEND SOURCE "${SRCS}")
-	list(APPEND SOURCE "${HDRS}")
-	list(APPEND SOURCE "Bindings/AllToLua.pkg")
-
-	includefolder("Resources")
-	source_group("" FILES ${SOURCE})
-
-	# Precompiled headers (1st part)
-	SET_SOURCE_FILES_PROPERTIES(
-		Globals.cpp PROPERTIES COMPILE_FLAGS "/Yc\"Globals.h\""
-	)
-	# CMake cannot "remove" the precompiled header flags, so we use a dummy precompiled header compatible with just this one file:
-	SET_SOURCE_FILES_PROPERTIES(
-		Bindings/Bindings.cpp PROPERTIES COMPILE_FLAGS "/Yc\"string.h\" /Fp\"$(IntDir)/Bindings.pch\""
-	)
-	list(APPEND SOURCE "Resources/Cuberite.rc")
-
-	# Make MSVC generate the PDB files even for the release build:
-	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /Zi")
-	set(CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_RELEASE}   /Zi")
-	set(CMAKE_EXE_LINKER_FLAGS_RELEASE    "${CMAKE_EXE_LINKER_FLAGS_RELEASE}    /DEBUG")
-	set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /DEBUG")
-	set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${CMAKE_MODULE_LINKER_FLAGS_RELEASE} /DEBUG")
```

Comment:
Should we be building RELEASE_WITH_DEBUG_INFO, or something? Do the windows stack traces still work?

Reply:
dun